### PR TITLE
Add EDITOR_GUIDELINES

### PR DESCRIPTION
**Pull request by Folyd**
I think we need guidelines for the editor to help us run Rust Magazine in good condition. Not only for existing editors, but also for new editors who may join us in the future.

Hence, I wrote a draft of EDITOR_GUIDELINES for Rust Magazine. Please give a review and comment if you have any questions. After we reach an agreement with all of the editors, I'll merge this PR. 

Thanks.

### CODE REVIEW

File: EDITOR_GUIDELINES.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+- Reviewing and editing articles submitted by authors
+- Assisting authors with improving the quality and clarity of their content
+- Ensuring that all articles are accurate, informative, and engaging
+- Collaborating with other editors and the Rust Magazine team to plan and execute editorial initiatives
+- Promoting Rust Magazine and its content on social media and other channels (prefer)
+- Staying up-to-date with the Rust community (prefer)
+
+## Joining
+
+If you are interested in joining the Rust Magazine editorial team, please fill this [Google form](https://xxx) with the following information:
```
Comment (Folyd) [inline EDITOR_GUIDELINES.md:29]: We need to update the Google form link after we prepare it. Cc. @LCrossman

**Comment by LCrossman** [comment]
Hi,  These look reasonable and good guidelines.  
I would just have one point about being responsible for accuracy of all articles could be a bit difficult, if they are not in an Editors knowledge areas, have some details in computer languages the Editor is not familiar with or have some sections written in a human language the Editor does not speak.

File: EDITOR_GUIDELINES.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+In the event that you need to step down from your role as an editor for Rust Magazine, we understand that life circumstances can change. We kindly request that you inform us of your intentions by sending PR to move your profile into alumni section.
+
+We are grateful for your contributions to Rust Magazine and would like to take this opportunity to thank you for your time and dedication to the publication. We wish you all the best in your future endeavors and hope that you will continue to be a part of the Rust community.
+
+## How to maintain Rust Magazine
+
+### Zine
+
+Rust Magazine is powered by [zine], a user-friendly static site generator designed specifically for magazines. Familiarity with zine is a valuable asset in maintaining the magazine as it offers built-in code blocks, such as the `urlpreview` to display an OGP preview of a URL, and `quote` to highlight someone's sentence, this is a good use case for an interview article. The repository and accompanying [documentation](https://zineland.github.io/) are great resources for getting started with zine.
```
Comment (mfrw) [inline EDITOR_GUIDELINES.md:45]:
```suggestion
Rust Magazine is powered by [zine](https://github.com/zineland/zine), a user-friendly static site generator designed specifically for magazines. Familiarity with zine is a valuable asset in maintaining the magazine as it offers built-in code blocks, such as the `urlpreview` to display an OGP preview of a URL, and `quote` to highlight someone's sentence, this is a good use case for an interview article. The repository and accompanying [documentation](https://zineland.github.io/) are great resources for getting started with zine.
```

File: EDITOR_GUIDELINES.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+### Tracking issue
+
+For each magazine issue, it is recommended to create a dedicated GitHub tracking issue, such as [#4]. This gives us a bird-eye on the progress of the current issue, and can be closed after the latest issue has been published.
```
Comment (mfrw) [inline EDITOR_GUIDELINES.md:49]: Nit:
```suggestion
For each magazine issue, it is recommended to create a dedicated GitHub tracking issue, such as [#4]. This gives us a birds-eye view on the progress of the current issue, and can be closed after the latest issue has been published.
```

**Review by mfrw — APPROVED** [review]
A few cosmetic nit-pics :)
LGTM otherwise.

File: EDITOR_GUIDELINES.md (hunk 2 above)
Comment (Folyd) [reply EDITOR_GUIDELINES.md:45]: We declared the zine link in line 103 😄

**Comment by ZhangHanDong** [comment]
LGTM

**Review by malagant — APPROVED** [review]
LGTM +1

**Comment by LCrossman** [comment]
added small change to responsibility section

**Review by LCrossman — APPROVED** [review]

**Review by pau1tuck — APPROVED** [review]

**Comment by Folyd** [comment]
ping @dmosc, @macko76

**Review by dmosc — APPROVED** [review]
LGTM!

**Comment by Folyd** [comment]
Hey, @macko76, do you mind reviewing this PR? Thanks.

**Pull request merged**